modules/500-openvpn/images/openvpn/entrypoint/entrypoint.go

```diff
@@ -0,0 +1,293 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"golang.org/x/sys/unix"
+	"log"
+	"net"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/coreos/go-iptables/iptables"
+	"github.com/vishvananda/netlink"
+)
+
+type iptablesRule struct {
+	Table string
+	Chain string
+	Rule  []string
+	Pos   int
+}
+
+func main() {
+	network, ok := os.LookupEnv("TUNNEL_NETWORK")
+	if !ok {
+		log.Fatal("The TUNNEL_NETWORK environment variable does not exist.")
+	}
+
+	protocol, ok := os.LookupEnv("OPENVPN_PROTO")
+	if !ok {
+		log.Fatal("The TUNNEL_NETWORK environment variable does not exist.")
+	}
+
+	var mgmtport string
+	var routeTable int
+
+	switch protocol {
+	case "tcp":
+		mgmtport = "8989"
+		routeTable = 10
+	case "udp":
+		mgmtport = "9090"
+		routeTable = 11
+	default:
+		log.Fatalf("OPENVPN_PROTO env value must be tcp or udp: %s", protocol)
+	}
+
+	iptablesMgr, err := iptables.NewWithProtocol(iptables.ProtocolIPv4)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	nat := iptablesRule{
+		Table: "nat",
+		Chain: "POSTROUTING",
+		Rule:  strings.Fields(fmt.Sprintf("-s %s ! -d %s -j MASQUERADE", network, network)),
+		Pos:   1,
+	}
+
+	manglePreroutingSetMark := iptablesRule{
+		Table: "mangle",
+		Chain: "PREROUTING",
+		Rule:  strings.Fields(fmt.Sprintf("-i tun-%s -j CONNMARK --set-mark %d", protocol, routeTable)),
+		Pos:   1,
+	}
+
+	manglePreroutingRestoreMark := iptablesRule{
+		Table: "mangle",
+		Chain: "PREROUTING",
+		Rule:  strings.Fields("! -i tun+ -j CONNMARK --restore-mark"),
+		Pos:   2,
+	}
+	mangleOutputRestoreMark := iptablesRule{
+		Table: "mangle",
+		Chain: "OUTPUT",
+		Rule:  strings.Fields("-j CONNMARK --restore-mark"),
+		Pos:   1,
+	}
+
+	err = insertUnique(iptablesMgr, nat.Table, nat.Chain, nat.Rule, nat.Pos)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	err = insertUnique(iptablesMgr, manglePreroutingSetMark.Table, manglePreroutingSetMark.Chain, manglePreroutingSetMark.Rule, manglePreroutingSetMark.Pos)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	err = insertUnique(iptablesMgr, manglePreroutingRestoreMark.Table, manglePreroutingRestoreMark.Chain, manglePreroutingRestoreMark.Rule, manglePreroutingRestoreMark.Pos)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	err = insertUnique(iptablesMgr, mangleOutputRestoreMark.Table, mangleOutputRestoreMark.Chain, mangleOutputRestoreMark.Rule, mangleOutputRestoreMark.Pos)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	rule := netlink.NewRule()
+	rule.Table = routeTable
+	rule.Priority = 1
+	rule.Mark = routeTable
+	err = netlink.RuleAdd(rule)
+
+	if err != nil {
+		if !os.IsExist(err) {
+			log.Fatal(err)
+		}
+	}
+
+	err = mknodDevNetTun()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	err = netLinkCreateTuntap(fmt.Sprintf("tun-%s", protocol), 1400)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	routeAdd(network, fmt.Sprintf("tun-%s", protocol), routeTable)
```

Comment:
please move  `fmt.Sprintf("tun-%s", protocol)` to variable and use this variable also in beyond

Reply:
done